Reference/Routing/Request-Pipeline/IContentFinder.md

```diff
@@ -91,6 +91,33 @@ this example shows how to add custom content finder to (and how to remove Conten
 
 #NotFoundHandlers
 
-To set your own 404 finder create an IContentFinder and set it as the ContentLastChanceFinder.  A ContentLastChanceFinder will always return a 404 status code. Example:
+To set your own 404 finder create an IContentFinder and set it as the ContentLastChanceFinder. A ContentLastChanceFinder will always return a 404 status code. This example creates a new implementation of the IContentFinder and checks whether the requested content could not be found by using the default `Is404` property presented in the `PublishedContentRequest` class.
+
+    public class My404ContentFinder : IContentFinder {
+    	public bool TryFindContent(PublishedContentRequest contentRequest) {
+            //logic to find your 404 page and set it to contentRequest.PublishedContent
+	     CultureInfo culture = null;
+            if (contentRequest.HasDomain) {
+                culture = CultureInfo.GetCultureInfo(contentRequest.UmbracoDomain.LanguageIsoCode);
+            }
+
+            //replace 'home_doctype_alias' with the alias of your homepage
+            IPublishedContent rootNode = UmbracoContext.Current.ContentCache.GetByXPath("//home_doctype_alias").FirstOrDefault(n => n.GetCulture().ThreeLetterWindowsLanguageName == culture.ThreeLetterWindowsLanguageName);
```

Comment:
Don't want to do difficult.  But could we do `GetByXPath("root/home_doctype_alias")`.  That is more performant than a full scan using `//`. It's just to avoid bad practices.